packages/data-stores/src/i18n/resolvers.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * External dependencies
+ */
+import { apiFetch } from '@wordpress/data-controls';
+import type { APIFetchOptions } from '@wordpress/api-fetch';
+
+import { setLocalizedLanguageNames } from './actions';
+
+export function* getLocalizedLanguageNames( locale: string ) {
+	const localizedLanguageNames = yield apiFetch( {
+		// Forcefully add `_locale` here because the `data` parameter will re-write it as
+		// just `locale` (no underscore). Context here:
+		// https://github.com/Automattic/wp-calypso/pull/46328#discussion_r515674976
+		url: `https://public-api.wordpress.com/wpcom/v2/i18n/language-names?_locale=${ locale }`,
```

Comment:
Would the rootURL be `https://public-api.wordpress.com/wpcom`? Perhaps we should open an issue for this refactor given it will touch other stores as well?

Reply:
I'm assuming the root URL would be `https://public-api.wordpress.com/` and `wpcom/v2` would actually be the namespace. If you inspect Calypso, you'll see plenty of requests to the same root URL, but for different namespaces - for example `rest/v1.1` or `rest/v1.3`.

But yep, an issue makes sense 👍 . By all means, I don't think it should block this PR anyway.

Reply:
https://github.com/Automattic/wp-calypso/issues/47675